rfcs/0003-timestamps-and-ttl.md

```diff
@@ -264,32 +251,79 @@ pub struct KeyValueMeta {
 
 ### Compaction
 
+#### TTL Compaction Filter
+
 We will introduce the concept of a compaction filter, which serves as a mechanism for pluggable
 participation in the compaction process. Specifically, the filter (if configured) will run on each
 key during the compaction process and optionally remove or modify the value.
 
 For the scope of this RFC, we will be introducing a builtin TS/TTL compaction filter. This filter
-will add a tombstone to any value where `db.clock.now() > kv.meta.ts + kv.meta.ttl`.
+will add a tombstone to any value where `db.clock.now() > kv.meta.ts + kv.meta.ttl`. Note that
+tombstones will still contain a timestamp see [out of order insertions](#insertion-timestamps).
 
 Since this RFC does not cover the public API for compaction filters, this filter will simply run
 on any SST that specifies `meta_features: [Timestamp | TimeToLive]`.
 
-There are various interesting interleaving scenarios with timestamps and TTL, outlined below:
-
-1. **(Happy Path)** KeyA is inserted at ts=100 with ttl=100. It will be retrievable (not filtered)
-   until ts=200 and will be replaced with a tombstone when compaction runs at ts>=200.
-2. **(In Order Overwrite, Newer TTL)** KeyA is inserted at ts=100 with ttl=100, then again at ts=150
-   with ttl=100. Any gets called after ts=150 will return the new value, and it will be accessible
-   until ts=250.
-3. **(In Order Overwrite, Older TTL)** KeyA is inserted at ts=100 with ttl=100, then again at ts=110
-   with ttl=40. Any gets called after ts=150 will not return any data (considered expired). If
-   compaction happens at ts=160, it will add a tombstone and propagate it down to overwrite the
-   original insert.
-4. **(Out of Order Overwrites)** KeyA is inserted at ts=100, then again at ts=50. Retrieval
-   information is outlined in the previous section depending on the `retrieval_grace_ms` configured
-   in the read options. Compaction will prefer the row with the larger timestamp. To guarantee
-   consistent time semantics, it is necessary to specify a sufficiently large value for the grace
-   period.
-5. **(Out of Order Tombstones)** KeyA is inserted at ts=100, then a delete for KeyA at ts=50 is
-   executed. The tombstone will be associated with ts=50, and therefore when the compactor merges
-   the tombstone and the existing key the tombstone will be ignored.
\ No newline at end of file
+#### Periodic Compaction
+
+If the SST is written with the `TimeToLive` meta feature enabled, the compaction scheduler will
+compare the SST's `SsTableInfo#timestamp` with a newly introduced compaction config for periodic
```

Comment:
Is this useful (or does it affect) GC at all? I don't think so, but wanted to make a note in case I missed anything.

Reply:
It could impact GC once we figure out the thing about having a table-wide TTL and expiring the entire SST at once (https://github.com/slatedb/slatedb/pull/228#discussion_r1771958947), though that's slightly orthogonal.